src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/HashSetBuilder.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+#pragma warning disable CA1825 // Avoid zero-length array allocations.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis
+{
+    // Based on the ArrayBuilder.cs implementation.
+    internal sealed class HashSetBuilder<T, TComparer>
```

Comment:
💡 If it's present in this target, `ImmutableSegmentedHashSet<T>.Builder` should suffice.